org/postgresql/xa/PGXAConnection.java

```diff
@@ -331,6 +332,12 @@ public int prepare(Xid xid) throws XAException {
         }
     }
 
+    private String currentDatabase() throws SQLException{
+    	DatabaseMetaData metaData = conn.getMetaData();
+    	Properties properties = Driver.parseURL(metaData.getURL(),new Properties());
```

Comment:
App server Transaction recovery process call this code to find pending transaction.
We use Jboss EAP 6x/AS7 which call this code on 2 minutes (Jboss AS/EAP default setting) interval. its good to catch database name on somewhere.